### PR TITLE
CLOUDP-278403: Spectral README update for linting/OAS changes

### PR DESCRIPTION
**Pull request by lovisaberggren**
## Proposed changes

Updating the README for spectral changes to guide MMS/openapi maintainers depending on the change to the linting/OAS and changes needed to propagate to MMS.

_Jira ticket: [CLOUDP-278403](https://jira.mongodb.org/browse/CLOUDP-278403)

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: tools/spectral/README.md [hunk 1]
```diff
@@ -1,11 +1,38 @@
...
+2. If the current `mongodb/openapi` spectral rules will violate the OAS changes, open a PR in `mongodb/openapi` and update/disable any rules that will fail
+3. Validate that the Spectral lint passes in `mongodb/openapi` and in MMS
+4. Review and merge both PRs
+5. Wait for the next release when the published OAS is updated
+6. Open a `mongodb/openapi` PR updating the linting `spectral-lint.yaml` with the linting changes initially added to MMS
+7. Validate that all tests pass
+8. Review and merge the PR
+9. Open a PR in MMS, updating the commit SHA of the imported spectral file, and removing any rules that were added to `mongodb/openapi`
+10. Validate all tests pass
+11. Review and merge the MMS PR
```
Comment (andreaangiolillo) [inline tools/spectral/README.md:36]: I am not sure I understood these steps. In the scenario the new rule breaks the release, we need to update the MMS OAS (or the other services OASes depending on where is the error), backport the fix to the vbranch and wait 1 days that the service is deployed in all the env (dev, qa, staging and prod). once that happens, you can open a PR agains mongodb/openapi and update the spectral rule

File: tools/spectral/README.md [hunk 2]
```diff
@@ -1,11 +1,38 @@
...
-1. [ ] Open a PR with the spectral changes and validate that the Spectral lint checks work.
-2. [ ] Review and merge the PR.
-3. [ ] Open a PR in mms, updating the commit sha of the spectral file imported.
-4. [ ] Validate all tests pass.
\ No newline at end of file
+
+1. Open a `mongodb/openapi` PR with the changes to `tools/spectral/.spectral.yaml`
+2. Validate that the new Spectral lint checks pass
+3. Review and merge the PR
+4. Open a PR in MMS, updating the commit SHA of the imported spectral file
```
Comment (wtrocki) [inline tools/spectral/README.md:18]: FYI: This repository is public. For public repositories, we keep internal documentation in wikis. 
This way you can put links and all information without being restricted in any way.

Comment (lovisaberggren) [reply tools/spectral/README.md:18]: Yes, though I think the current content is alright to keep here (also suggested by @blva to update this README). @andreaangiolillo Are you good with keeping the contents here, or would you prefer an internal wiki linked through a golink instead?

File: tools/spectral/README.md (hunk 1 above)
Comment (andreaangiolillo) [reply tools/spectral/README.md:36]: I agree with @wtrocki on the fact that this is a public repo and we should not probably talk about MMS. I think you could add a link in the readme to the internal wiki: https://wiki.corp.mongodb.com/pages/viewpage.action?pageId=306447071 and move these steps there. You can even add the link to the PR template to make sure the developer had a look to the wiki before making changes to spectral.yml. Up to you if you want to add the wiki to the readme or to the template file

Comment (lovisaberggren) [reply tools/spectral/README.md:36]: Alright sounds good, thanks!

File: tools/spectral/README.md (hunk 2 above)
Comment (lovisaberggren) [reply tools/spectral/README.md]: Went with moving the contents to the wiki and keeping the README lighter for public contributions, RFAL!

**Review by andreaangiolillo — APPROVED** [review]
LGTM. Thanks for all the changes!

**Pull request merged**